src/main/java/org/prebid/server/privacy/PrivacyAnonymizationService.java

```diff
@@ -0,0 +1,390 @@
+        }
+        return userAnonymizer;
+    }
+
+    private DeviceAnonymizer createTCFDeviceAnonymizer(boolean isMaskIp, boolean maskInfo, boolean isMaskGeo) {
+        final DeviceAnonymizer deviceAnonymizer = new DeviceAnonymizer(TCF);
+        if (isMaskIp) {
+            deviceAnonymizer.maskIp();
+        }
+        if (isMaskGeo) {
+            deviceAnonymizer.maskGeoCoordinates();
+        }
+        if (maskInfo) {
+            deviceAnonymizer.cleanIds();
+        }
+        return deviceAnonymizer;
+    }
+
+    private static Geo cleanGeo(Geo geo) {
+        return geo != null
+                ? geo.toBuilder()
+                .lat(null).lon(null)
+                .metro(null).city(null).zip(null)
+                .build()
+                : null;
+    }
+
+    private static Geo maskGeoCoordinates(Geo geo) {
+        return geo != null
+                ? geo.toBuilder()
+                .lat(maskGeoCoordinate(geo.getLat()))
+                .lon(maskGeoCoordinate(geo.getLon()))
+                .build()
+                : null;
+    }
+
+    /**
+     * Returns masked geo coordinate with rounded value to two decimals.
+     */
+    private static Float maskGeoCoordinate(Float coordinate) {
+        return coordinate != null ? Float.valueOf(ROUND_TWO_DECIMALS.format(coordinate)) : null;
+    }
+
+    class DeviceAnonymizer {
+
+        private final String enforcementPolicy;
+        private final Set<BiFunction<Device, Device.DeviceBuilder, Device.DeviceBuilder>> maskFunctions;
+        private final Set<String> logMessages;
+
+        DeviceAnonymizer(String enforcementPolicy) {
+            this.enforcementPolicy = enforcementPolicy;
+            this.maskFunctions = new HashSet<>();
+            this.logMessages = new HashSet<>();
+        }
+
+        public DeviceAnonymizer cleanIds() {
+            maskFunctions.add((device, deviceBuilder) -> deviceBuilder.ifa(null).macsha1(null).macmd5(null)
+                    .dpidsha1(null).dpidmd5(null).didsha1(null).didmd5(null));
+            logMessages.add(String.format("Device ids were removed form request to bidder according to %s policy.",
+                    enforcementPolicy));
+            return this;
+        }
+
+        private DeviceAnonymizer maskIp() {
+            maskFunctions.add((device, deviceBuilder) -> deviceBuilder.ip(ipAddressHelper.maskIpv4(device.getIp())));
+            maskFunctions.add((device, deviceBuilder) -> deviceBuilder.ipv6(ipAddressHelper
+                    .anonymizeIpv6(device.getIpv6())));
+            logMessages.add(String.format("Device IPs were masked in request to bidder according to %s policy.",
+                    enforcementPolicy));
+            return this;
+        }
+
+        public DeviceAnonymizer maskGeoCoordinates() {
+            maskFunctions.add((device, deviceBuilder) ->
+                    deviceBuilder.geo(PrivacyAnonymizationService.maskGeoCoordinates(device.getGeo())));
```

Comment:
The `PrivacyAnonymizationService` deals with TCF, CCPA and COPPA privacies. Why it cares about geo?